cmd/txt-prov/main.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"crypto"
+	"flag"
+	"fmt"
+
+	tss "github.com/9elements/go-tss"
+)
+
+func main() {
+	flag.Parse()
+	err := run()
+	if err != nil {
+		fmt.Printf("%v", err)
+		return
+	}
+}
+
+func run() error {
+	var err error
+	var delHash, writeHash []byte
+	var hA *crypto.Hash
+	if *help {
+		showHelp()
+		return nil
+	}
+	if *version {
+		showVersion()
+		return nil
+	}
+
+	tpmCon, err := tss.NewTPM()
+	if err != nil {
+		return err
+	}
+	defer tpmCon.Close()
+
+	switch tpmCon.Version {
+	case tss.TPMVersion12:
+		// Only uses SHA1, so we don't need -t evaluation
+		return fmt.Errorf("TPM 1.2 not supported yet")
+	case tss.TPMVersion20:
+		// Uses SHA256, SHA384, SHA512
+
+		if *loadfiles {
+			delHash, writeHash, _, hA, err = loadFiles()
+			if err != nil {
+				return err
+			}
+		} else {
+			hA, err := deconstructHashesAlg()
+			if err != nil {
+				return err
+			}
+			delHash, writeHash, err = handlePasswordsTPM20(*hA)
+			if err != nil {
+				return err
+			}
+			if err := writeFile("delHash", delHash, *hA); err != nil {
+				return err
+			}
+			if err = writeFile("writeHash", writeHash, *hA); err != nil {
+				return err
+			}
+		}
+
+		if !*deletePS && !*deleteAUX {
```

Comment:
looks like the 3 possible commands are DELETEPS, DELETEAUX, PROVISION. That should be the last argument you pass to this application and it must be mandatory.